### DISCUSSION TITLE
Renaming files with .sdr in docsettings folder

### DISCUSSION THREAD

**Discussion by ichnilatis-gr** [koreader/koreader#11109] [Q&A] [answered]
Hi!
I would like to ask: When we have the .sdr files in the docsettings folder and we rename a file or a folder, is that file or folder also renamed in the docsettings folder automatically? 
I read in the instructions that "any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books". Does this refer to reorganising files on a computer or also within KOReader?

Thank you!

**Comment by hius07** [comment, marked as answer]
We track most of file operations (rename/copy/move/delete) within our file browser and keep all the metadata and the History updated.
Except the case: metadata is stored in ./docsettings, and you rename a folder containing some books. Their metadata is not updated.

**Reply by hius07** [reply]
We do not track any file operations outside of KOReader.

**Reply by ichnilatis-gr** [reply]
If only books are renamed, are their metadata updated? i.e. are their .sdr files renamed?

**Reply by hius07** [reply]
No, sdr files are not renamed (and they mustn't be renamed).
sdr folders are renamed,

**Reply by hius07** [reply]
Well, if you change the file extension, sdr file is renamed too.

**Reply by ichnilatis-gr** [reply]
Sorry, that's what I meant. Hence, when we rename a book, its sdr folder is renamed?

**Reply by hius07** [reply]
Yes for "book folder" and "./docsettings" metadata locations, no for "./hashdocsettings".

**Reply by ichnilatis-gr** [reply]
Thank you for the clarification.

**Reply by hius07** [reply]
You're welcome.

**Reply by ichnilatis-gr** [reply]
Let me ask one more thing: If we rename a folder, can we rename the relative .sdr folder to the "./docsettings" folder manually, so that we keep the metadata?

**Reply by hius07** [reply]
What is the purpose? Please give an example.
("./docsettings" is an option in Gear-Documet-Book metadata location)

**Reply by ichnilatis-gr** [reply]
I mean, If I want to change the name of a folder, for various reasons, what have I to do so that I can preserve the metadata of the books of that folder? Can I change the name of the relative folder in the "./docsettings" folder manually?

**Reply by hius07** [reply]
Yes, if your setting of "Book metadata location" is "./docsettings", you can manually rename **the path** to sdr subfolder in koreader/docsettings.

**Reply by ichnilatis-gr** [reply]
Thanks!
